wgpu-hal/src/dx12/device.rs

```diff
@@ -328,6 +356,7 @@ impl crate::Device<super::Api> for super::Device {
         unsafe { queue.raw.destroy() };
     }
 
+    #[cfg(feature = "windows_rs")]
```

Comment:
This all looks good, but could we put all the allocator stuff in it's own module such that this code doesn't have to be cfg guarded. For example the non-cfg can have a memory allocator type that is just (). That kind of thing.